src/hotspot/share/runtime/interfaceSupport.inline.hpp

```diff
@@ -226,49 +227,58 @@ class ThreadToNativeFromVM : public ThreadStateTransition {
   }
 };
 
-
-// Parameter in_flight_mutex_addr is only used by class Mutex to avoid certain deadlock
-// scenarios while making transitions that might block for a safepoint or handshake.
-// It's the address of a pointer to the mutex we are trying to acquire. This will be used to
-// access and release said mutex when transitioning back from blocked to vm (destructor) in
-// case we need to stop for a safepoint or handshake.
-class ThreadBlockInVM : public ThreadStateTransition {
+template <typename PRE_PROC>
```

Comment:
:)

Reply:
Any reason why you don't use a ThreadClosure instead of the template parameter. I reckon it cannot be performance since it is only used on the slow path.